sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -153,6 +158,36 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+
+  test("CREATE VIEW IF NOT EXISTS never throws TABLE_OR_VIEW_ALREADY_EXISTS") {
+    // Concurrently create a view with the same name, so that some of the queries may all
+    // get that the view does not exist and try to create it. But with IF NOT EXISTS, the
+    // queries should not fail.
+    import ExecutionContext.Implicits.global
+    val concurrency = 10
+    val tableName = "table_name"
+    val viewName = "view_name"
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (id int) USING parquet")
+      withView("view_name") {
+        val futures = (0 to concurrency).map { _ =>
```

Comment:
the fix is trivial, we don't need to commit this fragile and heavy test. You can just run it off line and verify that the fix works.

Reply:
Yes I have verified that without the fix this test failed.